### PR TITLE
Add headers to timeline page sections 

### PR DESCRIPTION
**Pull request by kenzieschmoll**
These headers are consistent with other pages of DevTools (debugger, logging, etc.)
Screen Shot 2020-07-01 at 10 43 11 AM

The Timeline events header is slightly taller in preparation for adding a search bar

### CODE REVIEW

File: packages/devtools_app/lib/src/timeline/timeline_flame_chart.dart [hunk 1]
```diff
@@ -20,6 +21,97 @@ import '../utils.dart';
...
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    final newController = Provider.of<TimelineController>(context);
+    if (newController == controller) return;
+    controller = newController;
+  }
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/timeline/timeline_flame_chart.dart:57]: The code in this build method was moved from timeline_screen.dart. The areaPaneHeader addition and rearranging of the OutlineDecoration is the only change that is not copy paste.

File: packages/devtools_app/test/timeline_flame_chart_test.dart [hunk 2]
```diff
@@ -70,7 +71,7 @@ void main() {
       await tester.pumpAndSettle();
       expect(find.byType(TimelineFlameChart), findsNothing);
       expect(
-        find.byKey(TimelineScreen.emptyTimelineKey),
+        find.byKey(TimelineFlameChartContainer.emptyTimelineKey),
         findsOneWidget,
```
Comment (peterdjlee) [inline packages/devtools_app/test/timeline_flame_chart_test.dart:75]: nit: remove trailing comma for consistency or add trailing comma for all other `emptyTimelineKey` tests.

**Comment by peterdjlee** [comment]
LGTM with one nit

File: packages/devtools_app/test/timeline_flame_chart_test.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/test/timeline_flame_chart_test.dart]: doen

File: packages/devtools_app/test/timeline_flame_chart_test.dart [hunk 3]
```diff
@@ -60,7 +60,8 @@ void main() {
       await pumpTimelineBody(tester, TimelineController());
       await tester.pumpAndSettle();
       expect(find.byType(TimelineFlameChart), findsOneWidget);
-      expect(find.byKey(TimelineScreen.emptyTimelineKey), findsNothing);
+      expect(find.byKey(TimelineFlameChartContainer.emptyTimelineKey),
+          findsNothing);
```
Comment (terrylucas) [inline packages/devtools_app/test/timeline_flame_chart_test.dart:64]: nit: trailing comma

File: packages/devtools_app/test/timeline_flame_chart_test.dart [hunk 4]
```diff
@@ -69,10 +70,8 @@ void main() {
       await pumpTimelineBody(tester, TimelineController());
       await tester.pumpAndSettle();
       expect(find.byType(TimelineFlameChart), findsNothing);
-      expect(
-        find.byKey(TimelineScreen.emptyTimelineKey),
-        findsOneWidget,
-      );
+      expect(find.byKey(TimelineFlameChartContainer.emptyTimelineKey),
+          findsOneWidget);
```
Comment (terrylucas) [inline packages/devtools_app/test/timeline_flame_chart_test.dart:74]: nit trailing comma

File: packages/devtools_app/test/timeline_screen_test.dart [hunk 5]
```diff
@@ -85,7 +85,8 @@ void main() {
       await tester.pumpAndSettle();
       expect(find.byType(FlutterFramesChart), findsOneWidget);
       expect(find.byType(TimelineFlameChart), findsOneWidget);
-      expect(find.byKey(TimelineScreen.emptyTimelineKey), findsNothing);
+      expect(find.byKey(TimelineFlameChartContainer.emptyTimelineKey),
+          findsNothing);
```
Comment (terrylucas) [inline packages/devtools_app/test/timeline_screen_test.dart:89]: trailing comma

File: packages/devtools_app/test/timeline_screen_test.dart [hunk 6]
```diff
@@ -104,15 +105,17 @@ void main() {
       expect(controller.allTraceEvents, isNotEmpty);
       expect(find.byType(FlutterFramesChart), findsOneWidget);
       expect(find.byType(TimelineFlameChart), findsOneWidget);
-      expect(find.byKey(TimelineScreen.emptyTimelineKey), findsNothing);
+      expect(find.byKey(TimelineFlameChartContainer.emptyTimelineKey),
+          findsNothing);
```
Comment (terrylucas) [inline packages/devtools_app/test/timeline_screen_test.dart:109]: trailing comma here and others below.

**Pull request merged**